### ISSUE TITLE
Provide migration script from metronome

### ISSUE THREAD

**Issue by anubister** [anubister/prosody_ynh#9]
To ease migration from Metronome (default XMPP server integrated in Yunohost 11 core) while upgrading to Yunohost 12 with this app, provide a script migrating all expected datas : 

-  identify all the data to be migrated 
-  locate them in Metronome structure
- do the script
- test the script

See what initiated in 
https://github.com/YunoHost/yunohost/pull/240/files#diff-b5766a824ff7db64d714149acd51abe854017139664a1e46ada41c56daf522d9R16

to check if there is a newer version already covering the need?

**Comment by silverhook** [comment]
I just wanted to log some of my concerns / what I _hope_ the migration script could cover – the goal being that the XMPP users (not necessarily admins) would be negatively affected by the migration:

- automatic migration of accounts (I hope that’s a given)
- (automatic) migration of the rosters of all the accounts
- (automatic) migration of the chat history that’s on the server
- same for other stuff – think Movim, Libervia etc. that make use of XMPP PubSub
- (automatic) migration of connected services like Biboumi, Movim, Libervia, (hopefully Slidge soon), (I think Jitsi and PeerTube can also rely on XMPP)

**Comment by anubister** [comment]
> I just wanted to log some of my concerns / what I _hope_ the migration script could cover – the goal being that the XMPP users (not necessarily admins) would be negatively affected by the migration:
> 
>     * automatic migration of accounts (I hope that’s a given)
Yes, they are stored in the Yunohost core LDAP.

>     * (automatic) migration of the rosters of all the accounts
That should be easy:
metronome rosters are in /home/yunohost.app/metronome/domain/roster/myuser.dat , prosody are in 
/var/lib/prosody/domain/roster/myuser.dat ; they seem to have the same format.

Same for MUC: in private/myuser.dat for metronome as well as for prosody; they seem to have the same format.

>     * (automatic) migration of the chat history that’s on the server
offline messages are in offline/ directory for metronome as well as for prosody; they seem to have the same format.

>     * same for other stuff – think Movim, Libervia etc. that make use of XMPP PubSub
This remain to be checked as I dont use them. Could you check were are they located on your server?

>     * (automatic) migration of connected services like Biboumi, Movim, Libervia, (hopefully Slidge soon), (I think Jitsi and PeerTube can also rely on XMPP)
I think they should be any issue as these service are only pointing to the XMPP server. Or could you precise what do you expect?

**Comment by anubister** [comment]
Code provided by pitchum is working well https://github.com/anubister/prosody_ynh/blob/12c9f1ad238c8366e5c27d7de102db57b9b38313/scripts/install#L78C18-L78C27 (tested on roster and MUC bookmark)

I would propose the following UX:
* before install, check for metronome data (either from /var/lib/metronome (yunohost 11, so it should be the case only for strange migrations) or (with priority) from /home/yunohost.app/metronome/ (new data path for metronome app on ynh12))
  * if there are no prosody data, *propose* to *copy* those from metronome
  * if there are already prosody data, propose : 
     * to ignore metronome data to keep prosody data
     * to abort installation to manually check/manage which one are to be kept

before we should freeze the data location for this app (rather /home/yunohost.app/prosody as /var/lib/prosody).

**Comment by anubister** [comment]
Optional import is now implemented in `install-from-debian` , remaining bug tracked by #12

**Comment by silverhook** [comment]
> >    * same for other stuff – think Movim, Libervia etc. that make use of XMPP PubSub
> 
> This remain to be checked as I dont use them. Could you check were are they located on your server?

I think the easy answer is that Movim (and Libervia?) doesn’t seem to run on Metronome anyway, so  this scenario is moot.

> >    * (automatic) migration of connected services like Biboumi, Movim, Libervia, (hopefully Slidge soon), (I think Jitsi and PeerTube can also rely on XMPP)
> 
> I think they should be any issue as these service are only pointing to the XMPP server. Or could you precise what do you expect?

I only use Biboumi (for now), so that’s not something I can test much, I’m afraid.

I did remember another thing that is of concern to me:

- I don’t use my YunoHost (i.e. landing page) (sub)domain for XMPP, but a specific (sub)domain – I assume this is supported by Prosody; but on on the built-in Metronome I had to manually set up some e-mail aliases and some other settings to get that
- ideally this could be done in the WebUI
- ideally individual users could have different aliases, if they wanted to

**Comment by anubister** [comment]
> I did remember another thing that is of concern to me:
>     * I don’t use my YunoHost (i.e. landing page) (sub)domain for XMPP, but a specific (sub)domain – I assume this is supported by Prosody; 

Yes, you can choose the domain you want at the beginning of the installation.

> but on on the built-in Metronome I had to manually set up some e-mail aliases and some other settings to get that
>     * ideally this could be done in the WebUI
>     * ideally individual users could have different aliases, if they wanted to

This might be the case also here: it depend on which domain you selected when you created your users. Anyway, you can check and correct this by adding an email alias on the WebUI
![grafik](https://github.com/user-attachments/assets/253a0a89-7867-4e7e-a16f-90ed92272a64)

**Comment by silverhook** [comment]
Awesome, thanks!

It sounds like Prosody would be easier to set up than Metronome was for my use case. I hope that is the case :)

**Comment by Neustradamus** [comment]
@anubister: Good job, and good project to do it.

I am at initiative of Metronome IM into YunoHost.

A long time ago, I have informed @Maranda that with the Metronome IM less development, it will be switched to Prosody IM.

Linked to:
- https://github.com/YunoHost/yunohost/pull/1165
- https://github.com/YunoHost/doc/pull/473
- https://github.com/YunoHost/yunohost/pull/240
- https://github.com/YunoHost/install_script/pull/45
- https://github.com/YunoHost/issues/issues/67

**Comment by Neustradamus** [comment]
Linked to:
- https://github.com/YunoHost/yunohost/pull/1165
- https://github.com/YunoHost/doc/pull/473
- https://github.com/YunoHost/yunohost/pull/240
- https://github.com/YunoHost/install_script/pull/45

**Comment by anubister** [comment]
Prefer the migration via the new Prosody module https://modules.prosody.im/mod_storage_metronome_readonly ?

**Comment by Neustradamus** [comment]
@anubister: Good job from @linkmauve, the author of the Prosody module ^^

@anubister: It will be nice to verify if it is perfect.

Badly @alexAubin does not like when I specify your good work:
- https://github.com/YunoHost-Apps/prosody_ynh/pull/28#issuecomment-2559709289

**Comment by Neustradamus** [comment]
@anubister: Good job from @linkmauve, the author of the Prosody module ^^

@anubister: It will be nice to verify if it is perfect.

Badly @alexAubin does not like when I specify your good work:
- https://github.com/YunoHost-Apps/prosody_ynh/pull/28#issuecomment-2559709289

To recall: I promote, I have helped and I have installed several YunoHost machines since the beginning.
From 2012-06-02 (13 years in 2025):
- https://x.com/neustradamus/status/208988897848471553

**Comment by alexAubin** [comment]
@Neustradamus : Maybe @alexAubin would give a shit if you didn't got ban from the projects organization after randomly harassing the team with nonsense. Just like you got banned from other projects because of the same attitude. And then even now you manage to keep being annoying by randomly pinging me from an external repo. Jesus fucking christ.
